### ISSUE TITLE
How to fine-tune DeepFloyd IF?

### ISSUE THREAD

**Issue by xingzhehe** [huggingface/diffusers#3356]
It might be related to #3307.
I would like to try DreamBooth on DeepFloyd IF, but I found its pipeline seems slightly different from Stable Diffusion Pipeline.
https://github.com/huggingface/diffusers/blob/eade4308dabc7f7ba75eab508d386b66b3764513/src/diffusers/pipelines/deepfloyd_if/pipeline_if_img2img.py#L927
In addition to the noise, it additionally predicts the variance. Is it also used in the loss?

**Comment by patrickvonplaten** [comment]
@williamberman got it working in https://github.com/huggingface/diffusers/pull/3360 I think :-)

**Comment by williamberman** [comment]
hey @xingzhehe see this comment https://github.com/huggingface/diffusers/issues/3307#issuecomment-1540533807 :)

**Comment by jmaccall316** [comment]
> Thanks! I just tested and it works great! Do you have any plan to take variance into consideration? @williamberman

Hello, I am going to try this, just wondering if you had to scale your dataset to 64 or if it will auto scale any 1.1 resolution?

**Comment by oss-roettger** [comment]
I have just published an Open Source version of Textual Inversion to incorporate your own objects, faces, logos or styles into
DeepFloyd IF: https://github.com/oss-roettger/T5-Textual-Inversion
The clear and short notebook runs within 24 GB CUDA memory. Feedback welcome!
Kind regards
Hans

**Comment by williamberman** [comment]
hey @oss-roettger that's awesome! would you be interested in opening a PR adding to the diffusers scripts? I previously tried doing textual inversion with IF but the gradients that reached the embeddings were very small so we got little to no updates from the embedding for the initial token. did you run into the same problem and find a way around it or did it work out of the box for you?

**Comment by oss-roettger** [comment]
hey @williamberman, you are right, the gradients are very small. I found some indications in the IF source code that they multiplied the loss by a factor of 1000 (diffusion.num_timesteps) before applying it. But even then you need very high learning rates (0.5 -0.1) to see an effect. [This](https://github.com/oss-roettger/T5-Textual-Inversion/blob/main/T5_Inversion.ipynb) worked for me:
   
        # we will use the diffusion.training_losses function
        diffusion=if_I.get_diffusion(timestep_respacing='')  
        num_timesteps=diffusion.num_timesteps
        scale=torch.tensor(num_timesteps,dtype=torch.float).to(if_I.device)
        ...
        loss=diffusion.training_losses(if_I.model,sample,timesteps,if_I_kwargs)['loss']*scale

Feel free to use it for a diffusers PR.

**Comment by williamberman** [comment]
oh super cool! thank you so much for the pointer :)

**Comment by Yashvardhan31** [comment]
Is there any code for vanilla finetuning of DeepFloyd IF?

**Comment by cfeng16** [comment]
Same question above ^^ Any code to finetune DeepFloyd IF directly?
